Use github-webhook-validator for authentication

mbland opened the pull request:
Depends on #4. (At time of inception, this PR is only one commit ahead.)

The `secretKeyFile` configuration variables and `github-webhook-validator` component now allow us to define secret keys that can be used to authenticate incoming payloads before they're even parsed. This is of increasing concern as Pages continues to grow in popularity and visibility.

cc: @arowla @afeld

Commit:
Use github-webhook-validator for authentication

The `secretKeyFile` configuration variables and `github-webhook-validator`
component now allow us to define secret keys that can be used to authenticate
incoming payloads before they're even parsed. This is of increasing concern as
Pages continues to grow in popularity and visibility.

Commit:
Bump github-webhook-validator to v0.1.0

mbland commented:
OK, rebased this now that #4 is in. Happy to look for other reviewers if y'all're tapped.

afeld commented on index.js:
```diff
@@ -16,10 +16,19 @@ exports.versionString = function() {
+  return webhookValidator.loadKeyDictionary(
+    config.secretKeyFile, config.builders)
+    .then(function(keyDictionary) { return doLaunch(config, keyDictionary); })
+    .catch(function(err) { console.error('Failed to start server:', err); });
```
Minor: How about expanding these callbacks to multiple lines?

https://twitter.com/mixonic/status/656935651586256896

mbland replied:
Done.

afeld commented on pages-config.json:
```diff
@@ -14,6 +14,7 @@
   "assetRoot":        "/guides-template",
   "fileLockWaitTime": 30000,
   "fileLockPollTime": 1000,
+  "secretKeyFile":    "/home/ubuntu/.18f-pages.secret",
```
Not a blocker for this PR, but it's probably worth distinguishing the 18F-specific configuration from the general files/components, e.g. by moving it to a different repository, or at least making a note of it in the README.

mbland replied:
This is actually here to generate an initial template via `bin/18f-pages print-template`. The "real" config is in 18F/pages.

afeld commented on test/index.js:
```diff
@@ -0,0 +1,186 @@
+    req.write(payload);
+    req.end();
+  });
+}
```
Minor: Maybe worth moving these helper methods to a standalone file, so that this file is purely the tests themselves, for readability's sake.

mbland replied:
Done.

afeld commented on test/index.js:
```diff
@@ -0,0 +1,186 @@
+      port = server.address().port;
+      restoreLogs();
+      done();
+    }).catch(function(err) { restoreLogs(); done(err); });
```
Curious if you came across https://github.com/visionmedia/supertest? Might make some of this test setup easier (by outsourcing it).

mbland replied:
Hmm, I don't know if it could help with the setup... Maybe with sending the request, though I'm not 100% certain. Definitely going to study that framework some more, though; never heard of it before.

afeld commented:
Just a handful of minor notes. Otherwise, :ship: !

Commit:
Incorporate @afeld's feedback from #5

Namely, split single-line function definitions across lines and extracted a
new RequestHelper class to trim down the test/index.js file.

mbland commented:
Thanks for the review, @afeld! Gonna merge now.
